jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java

```diff
@@ -105,80 +106,95 @@ public final class MenuFactory {
 
   private static Set<String> classesToSkip() {
     return Arrays.stream(JMeterUtils.getPropDefault("not_in_menu", "").split(","))
-        .map(String::trim)
-        .collect(Collectors.toSet());
+            .map(String::trim)
+            .collect(Collectors.toSet());
   }
 
   private static void initializeMenus(
-      Map<String, List<MenuInfo>> menus, Set<String> elementsToSkip) {
+          Map<String, List<MenuInfo>> menus, Set<String> elementsToSkip) {
     try {
-      List<String> guiClasses = ClassFinder
-          .findClassesThatExtend(
-              JMeterUtils.getSearchPaths(),
-              new Class[]{JMeterGUIComponent.class, TestBean.class})
-          .stream()
-          // JMeterTreeNode and TestBeanGUI are special GUI classes,
-          // and aren't intended to be added to menus
-          .filter(name -> !name.endsWith("JMeterTreeNode"))
-          .filter(name -> !name.endsWith("TestBeanGUI"))
-          .filter(name -> !name.equals("org.apache.jmeter.gui.menu.StaticJMeterGUIComponent"))
-          .filter(name -> !elementsToSkip.contains(name))
-          .distinct()
-          .map(String::trim)
-          .collect(Collectors.toList());
+      List<String> guiClasses = getModifiedStrings(elementsToSkip);
```

Comment:
As described in the class javadoc, this class is just a copy of an existing class from JMeter. In such classes that contain so many lines from original code, we avoid doing much refactor as to keep the class as similar to the original one and ease porting changes in the original code to the "modified" version of the class.

In any case, this class was added because we were experiencing some flakiness in some tests and modifying this class would allow us to better trace the issue. The tests have been stable for a long time now, so I think is time we just remove this class.